subsys/net/lib/openthread/platform/uart.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2020 Tridonic GmbH & Co KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_OPENTHREAD_LOG_LEVEL
+#define LOG_MODULE_NAME net_otPlat_uart
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <kernel.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <drivers/uart.h>
+
+#include <sys/ring_buffer.h>
+#include <sys/atomic.h>
+
+#ifdef CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_ACM
+#include <usb/usb_device.h>
+#endif
+
+#include <openthread/platform/uart.h>
+
+#include "platform-zephyr.h"
+
+const struct openthread_uart {
+	const struct ring_buf *tx_ringbuf;
+	const struct device *dev;
+	atomic_t tx_busy;
+};
+
+#define OT_UART_DEFINE(_name, _tx_ringbuf_size)	\
+	RING_BUF_DECLARE(_name##_tx_ringbuf, _tx_ringbuf_size);		\
+	static const struct openthread_uart _name = {			\
+		.tx_ringbuf = &_name##_tx_ringbuf,			\
+	}
+
+OT_UART_DEFINE(ot_uart, 4096)
+
+enum {
+	kReceiveFifoSize = 128,
+};
+
+static void uart_rx_handle(const struct openthread_uart *ot_uart)
+{
+	u8_t data[kReceiveFifoSize];
+	u32_t rd_len;
+
+	rd_len = uart_fifo_read(ot_uart->dev,
+				data, kReceiveFifoSize);
+	if (rd_len > 0)
+		otPlatUartReceived(data, rd_len);
+}
+
+static void uart_tx_handle(const struct openthread_uart *ot_uart)
+{
+	const struct device *dev = ot_uart->dev;
+	u32_t len;
+
+	const u8_t *data;
+
+	len = ring_buf_get_claim(ot_uart->tx_ringbuf, (u8_t **)&data,
+				 ot_uart->tx_ringbuf->size);
+	if (len) {
+		int err;
+
+		len = uart_fifo_fill(dev, data, len);
+		err = ring_buf_get_finish(ot_uart->tx_ringbuf, len);
+		__ASSERT_NO_MSG(err == 0);
+	} else {
+		uart_irq_tx_disable(dev);
+		ot_uart->tx_busy = 0;
+	}
+
+	otPlatUartSendDone();
+}
+
+static void uart_callback(const void *user_data)
+{
+	const struct device *dev = ot_uart->dev;
+
+	while (uart_irq_update(dev) && uart_irq_is_pending(dev)) {
+
+		if (uart_irq_rx_ready(dev))
+			uart_rx_handle(ot_uart);
+
+		if (uart_irq_tx_ready(dev))
+			uart_tx_handle(ot_uart);
+	}
+};
+
+otError otPlatUartEnable(void)
+{
+	(&ot_uart)->dev = device_get_binding(
```

Comment:
Why `(&ot_uart)->dev` and not simply `ot_uart.dev`?

Reply:
Should be fixed now.